djangocms_navigation/views.py

```diff
@@ -29,24 +42,17 @@ def dispatch(self, request, *args, **kwargs):
             raise PermissionDenied
         return super().dispatch(request, *args, **kwargs)
 
-    def get_queryset(self):
+    def get_data(self):
         content_type_id = self.request.GET.get("content_type_id", None)
-
         query = self.request.GET.get("query", None)
         site = self.request.GET.get("site")
+
         try:
             content_object = ContentType.objects.get_for_id(content_type_id)
         except ContentType.DoesNotExist:
-            raise ValueError(
-                "Content type with id {} does not exists.".format(content_type_id)
-            )
+            return HttpResponseBadRequest()
```

Comment:
Aren't you already doing all these checks in the `get` method now? Would be good to keep all these validation checks in one place rather than spread across two methods.